velox/connectors/hive/iceberg/IcebergPartitionNameGenerator.h

```diff
@@ -0,0 +1,136 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/connectors/hive/PartitionNameGenerator.h"
+#include "velox/connectors/hive/iceberg/PartitionSpec.h"
+
+namespace facebook::velox::connector::hive::iceberg {
+
+/// Generates Iceberg-compliant partition path names.
+/// Converts partition keys to human-readable strings based on their transform
+/// types (e.g., year, month, day, hour, identity, truncate) and constructs
+/// URL-encoded partition paths in the format "key1=value1/key2=value2/...".
+class IcebergPartitionNameGenerator : public PartitionNameGenerator {
+ public:
+  /// @param partitionSpec Iceberg partition specification containing transform
+  /// definitions for each partition field. Used to create formatters that
+  /// convert transformed partition keys to human-readable strings based on
+  /// their transform types.
+  /// @param partitionPathAsLowerCase Whether to convert partition keys to
+  /// lowercase in the generated partition path. When true, partition keys like
+  /// "Year" become "year" in the path "year=2025/...".
+  IcebergPartitionNameGenerator(
+      const IcebergPartitionSpecPtr& partitionSpec,
+      bool partitionPathAsLowerCase);
+
+  /// Generates an Iceberg-compliant partition path string for the given
+  /// partition ID.
+  ///
+  /// Constructs a partition path in the format "key1=value1/key2=value2/..."
+  /// where:
+  /// - Keys are partition column names for identity transforms, or
+  ///   "columnName_transformName" for non-identity transforms (e.g.,
+  ///   "date_year")
+  /// - Values are human-readable string representations of the transformed
+  ///   partition keys, formatted according to their transform types
+  /// - Both keys and values are URL-encoded per java.net.URLEncoder.encode()
+  ///
+  /// Example: "store_id=123/date_year=2025/date_month=2025-01"
+  ///
+  /// Typically called once per partition ID when creating a new writer for that
+  /// partition (e.g., in HiveDataSink::appendWriter).
+  ///
+  /// @param partitionId Sequential partition ID (0-based) used as the row index
+  /// into partitionValues. Must be less than partitionValues->size().
+  /// @param partitionValues RowVector containing transformed partition keys
+  /// for all partitions. Each row represents one unique partition, with
+  /// columns corresponding to partition fields in partitionSpec. Row at
+  /// partitionId contains the keys for this specific partition.
+  /// @return URL-encoded partition path string suitable for use in file paths.
+  std::string partitionName(
+      uint32_t partitionId,
+      const RowVectorPtr& partitionValues) const override;
+
+  class IcebergPartitionValueFormatter : public PartitionNameGenerator {
```

Comment:
It is confusing to have IcebergPartitionNameGenerator class extend PartitionNameGenerator and contain IcebergPartitionValueFormatter class which also extends PartitionNameGenerator.